pkg/controller/podautoscaler/horizontal.go

```diff
@@ -171,9 +171,15 @@ func NewHorizontalController(
 
 	hpaInformer.Informer().AddEventHandlerWithResyncPeriod(
 		cache.ResourceEventHandlerFuncs{
-			AddFunc:    hpaController.enqueueHPA,
-			UpdateFunc: hpaController.updateHPA,
-			DeleteFunc: hpaController.deleteHPA,
+			AddFunc: func(obj interface{}) {
+				hpaController.enqueueHPA(ctx, obj)
```

Comment:
Passing a logger is preferred unless there are reasons to pass a context, like having to support cancellation. That's not applicable here.

Advantages of a logger:
- Lookup only done once.
- Clear semantic of the parameter.